File: ling-programacao/c/lista-1/ex02.c
#include <stdio.h>

int main() {
    int n;
    printf("Digite um número inteiro positivo: ");
    scanf("%d", &n);

    if (n <= 1) {
        printf("%d não é primo.\n", n);
        return 0;
    }

    int i, primo = 1;

    for (i = 2; i <= n / 2; ++i) {
        if (n % i == 0) {
            primo = 0;
            break;
        }
    }

    if (primo == 1)
        printf("%d é primo.\n", n);
    else
        printf("%d não é primo.\n", n);

    return 0;
}

File: ling-programacao/go/lista-1/ex02.go
package main

import (
    "fmt"
)

func Primo(n int) bool {
    if n <= 1 {
        return false
    }
    for i := 2; i*i <= n; i++ {
        if n % i == 0 {
            return false
        }
    }
    return true
}

func main() {
    n := 29
    if Primo(n) {
        fmt.Printf("%d é primo\n", n)
    } else {
        fmt.Printf("%d não é primo\n", n)
    }
}
